src/core/border_router/routing_manager.cpp

```diff
@@ -619,48 +705,37 @@ void RoutingManager::EvaluateNat64Prefix(void)
 {
     OT_ASSERT(mIsRunning);
 
-    NetworkData::Iterator            iterator = NetworkData::kIteratorInit;
-    NetworkData::ExternalRouteConfig config;
-    Ip6::Prefix                      smallestNat64Prefix;
+    Ip6::Prefix                      nat64Prefix;
+    RoutePreference                  routePreference;
+    Error                            error;
+    NetworkData::ExternalRouteConfig preferredNat64PrefixConfig;
+    bool                             should_advertise;
 
     LogInfo("Evaluating NAT64 prefix");
 
-    smallestNat64Prefix.Clear();
-    while (Get<NetworkData::Leader>().GetNextExternalRoute(iterator, config) == kErrorNone)
-    {
-        const Ip6::Prefix &prefix = config.GetPrefix();
+    VerifyOrExit(GetNat64Prefix(nat64Prefix, &routePreference) == kErrorNone);
+    error = Get<NetworkData::Leader>().GetPreferredNat64Prefix(preferredNat64PrefixConfig);
 
-        if (config.mNat64 && prefix.IsValidNat64())
-        {
-            if (smallestNat64Prefix.GetLength() == 0 || prefix < smallestNat64Prefix)
-            {
-                smallestNat64Prefix = prefix;
-            }
-        }
-    }
+    // NAT64 prefix is expected to be advertised from this BR when
+    // - no NAT64 prefix exits in Network Data yet
+    // - the preferred NAT64 prefix in Network Data was advertised by this BR
+    // TODO: change to check RLOC16 to determine if the NAT64 prefix was advertised by thiss BR
```

Comment:
```suggestion
    // TODO: change to check RLOC16 to determine if the NAT64 prefix was advertised by this BR
```

I think this should not be a TODO.

Reply:
I will update it in a follow-up PR.